crates/client/rpc/src/lib.rs

```diff
@@ -847,12 +848,21 @@ where
         } else {
             starknet_block.transactions_hashes::<H>(chain_id.0.into()).map(FieldElement::from).collect()
         };
+        let block_status = match self.backend.messaging().last_synced_l1_block_with_event() {
+            Ok(l1_block) => {
+                if l1_block.block_number >= starknet_block.header().block_number {
+                    BlockStatus::AcceptedOnL1
+                } else {
+                    BlockStatus::AcceptedOnL2
+                }
+            },
+            Err(e) => Err(anyhow!("couldn't retrieve block from l1 : {}", e)) ?,
```

Comment:
we should probably do the same as above here i.e return a CallError::Failed
```
CallError::Failed(anyhow::anyhow!(
                        "The hash cached for block with hash {block_hash:?} is an invalid felt: '{tx_hash}'. The \
                         caching db has probably been tempered"
                    ))
                
```

Reply:
done